builtin/providers/aws/resource_aws_codecommit_trigger_test.go

```diff
@@ -0,0 +1,160 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSCodeCommitTrigger_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCodeCommitTriggerDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitTriggerExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSCodeCommitTrigger_withChanges(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCodeCommitTriggerDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitRepositoryExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.794783505.name", "noname"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_withChanges,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitRepositoryExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.807285978.name", "noname_withChanges"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCodeCommitTriggerDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).codecommitconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_codecommit_trigger" {
+			continue
+		}
+
+		_, err := conn.GetRepositoryTriggers(&codecommit.GetRepositoryTriggersInput{
+			RepositoryName: aws.String(rs.Primary.ID),
+		})
+
+		if ae, ok := err.(awserr.Error); ok && ae.Code() == "RepositoryDoesNotExistException" {
```

Comment:
Can we add the same check to the REad func in the main resource? This seems a valid error to guard against